src/api/app/controllers/webui/obs_factory/staging_projects_controller.rb

```diff
@@ -0,0 +1,65 @@
+module Webui::ObsFactory
+  class StagingProjectsController < ApplicationController
+    respond_to :json, :html
+
+    before_action :require_distribution
+    before_action :require_project_name, only: [:show]
+
+    def index
+      respond_to do |format|
+        format.html do
+          @staging_projects = ::ObsFactory::StagingProjectPresenter.sort(@distribution.staging_projects_all)
+          @backlog_requests = ::ObsFactory::Request.with_open_reviews_for(by_group: @distribution.staging_manager, target_project: @distribution.name)
+          @requests_state_new = ::ObsFactory::Request.in_state_new(by_group: @distribution.staging_manager, target_project: @distribution.name)
+
+          staging_project = Project.find_by_name("#{@distribution.project}:Staging")
+
+          file = ::Backend::Api::Sources::Package.file(staging_project.name, 'dashboard', 'ignored_requests')
+          unless file.to_s.nil?
+            @ignored_requests = YAML.load(file.to_s)
+          end
+          if !@ignored_requests.nil? and @ignored_requests
```

Comment:
:scream:

Reply:
This code needs to be rewritten completely in the future anyway as this should move into an obs attribute. Polishing this old functionality to perfection is beyong logic.

Reply:
The scarier part is the need for `unless file.to_s.nil?`...`to_s.nil?`

Reply:
It's actually no longer needed. Your code was using PackageFile.new - which has a rather strange API as it tries to mimik ActiveRecord::Model.

But @DavidKang changed it to use the file method - and that one simply returns a string or throws an exception.

Reply:
> jberry-suse  a day ago
> The scarier part is the need for unless file.to_s.nil?...to_s.nil?

I was also suprised... We [renamed](https://github.com/openSUSE/open-build-service/pull/5183/files#diff-9755b746898a16e5abe51d75c3cecf87R59) it now